app/scenes/Document/components/Header.js

```diff
@@ -65,17 +77,139 @@ function DocumentHeader({
   headings,
 }: Props) {
   const { t } = useTranslation();
-  const { auth, ui, policies } = useStores();
+  const { auth, ui, policies, collections, documents } = useStores();
   const isMobile = useMobile();
+  const { showToast } = useToasts();
+  const dialog = useDialogState({ modal: true });
+  const theme = useTheme();
+  const hasCollection = collections.get(document.computedCollectionId);
+  const [searchTerm, setSearchTerm] = React.useState();
+  const [selectedPath, setSelectedPath] = React.useState<?DocumentPath>();
+
+  const checked = React.useCallback(
+    (result) => {
+      if (!selectedPath) return;
+
+      if (selectedPath.type === "collection" && selectedPath.id === result.id) {
+        return true;
+      }
+      if (
+        selectedPath.type === "document" &&
+        selectedPath.id === result.id &&
+        selectedPath.collectionId === result.collectionId
+      ) {
+        return true;
+      }
+      return false;
+    },
+    [selectedPath]
+  );
 
   const handleSave = React.useCallback(() => {
     onSave({ done: true });
   }, [onSave]);
 
   const handlePublish = React.useCallback(() => {
+    if (!hasCollection) {
+      dialog.setVisible(true);
+      return;
+    }
     onSave({ done: true, publish: true });
-  }, [onSave]);
+  }, [dialog, hasCollection, onSave]);
+
+  const handleFilter = (ev) => {
+    setSearchTerm(ev.target.value);
+  };
+
+  const searchIndex = React.useMemo(() => {
```

Comment:
Way too much going on this component, please refactor the popover into it's own component 😄